.sandbox/provision/lifecycle-asg/cleanup.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# cleanup leaked instances
+echo "Checking and cleaning up the leaked instances (if any)"
+INSTANCE_IDS="$(aws ec2 describe-instances --filters "Name=instance-state-name,Values=running" "Name=tag:sandbox-asg,Values=true" --query 'Reservations[].Instances[?!(Tags[?Key==`aws:autoscaling:groupName`] || Tags[?Key==`SandboxID`])].InstanceId[]')"
+
+while IFS= read -r instance_id; do
```

Comment:
For `IFS`, the default is spacing, I don't think `IFS=` is needed here. Also please use `INSTANCE_ID` for naming convention (this is a global variable).